### PR TITLE
enh(ci): Adding cypress e2e test parallelization on CI/CD

### PR DESCRIPTION
**Pull request by SoldierBoy31**
## Description

Updating github workflow to run multiple E2E tests in parallel. This will reduce build time.

## Type of change

- [X] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [X] 23.04.x (master)

<h2> How this pull request can be tested ? </h2>

Run CI/CD

#### Community contributors & Centreon team

- [X] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [X] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: .github/workflows/web.yml [hunk 1]
```diff
@@ -376,27 +376,17 @@ jobs:
...
-      - name: Login to Registry
-        uses: docker/login-action@v2
+      - name: Cypress e2e testing
+        uses: ./.github/workflows/cypress-e2e-parrallelization.yml
         with:
-          registry: docker.centreon.com
+          name: e2e
+          module_name: centreon
+          image_name: docker.centreon.com/${{ inputs.image_name }}-centos7:${{ github.head_ref || github.ref_name }}
+          features_path: features
```
Comment (Thebarda) [inline .github/workflows/web.yml:385]:
```suggestion
          features_path: cypress/e2e
```

File: .github/actions/cypress-e2e-testing/action.yml [hunk 2]
```diff
@@ -26,13 +28,14 @@ runs:
     - name: "Start Centreon container"
       run: docker run -p 4000:80 -d -v "/var/run/docker.sock:/var/run/docker.sock" --name e2e-tests-${{ inputs.module }} ${{ inputs.image }}:${{ inputs.image_version }}
       shell: bash
-
+
```
Comment (Thebarda) [inline .github/actions/cypress-e2e-testing/action.yml:31]:
```suggestion

```

**Review by Thebarda — COMMENTED** [review]
It might be better to change the name of the videos archive following the spec

File: .github/actions/cypress-e2e-testing/action.yml [hunk 3]
```diff
@@ -26,20 +28,20 @@ runs:
...
         working-directory: ${{ inputs.module }}/tests/e2e
+        spec: cypress/e2e/**/${{ inputs.path_spec }}
 
     - name: Archive videos
       uses: actions/upload-artifact@v3
       if: ${{ failure() }}
       with:
         name: cypress-videos
-        path: ${{ inputs.module }}/tests/e2e/cypress/results/videos
+        path: ${{ inputs.module }}/tests/e2e/cypress/results/videos/${{ inputs.path_spec }}/
```
Comment (Thebarda) [inline .github/actions/cypress-e2e-testing/action.yml:44]: I meant about the name property rather than the path :)

Comment (SoldierBoy31) [reply .github/actions/cypress-e2e-testing/action.yml:44]: Like this for example: cypress-videos-${{ inputs.path_spec }} ?

File: .github/actions/cypress-e2e-testing/action.yml [hunk 4]
```diff
@@ -10,14 +10,16 @@ inputs:
   module:
     description: "module"
     required: true
+  path_spec:
```
Comment (bdauria) [inline .github/actions/cypress-e2e-testing/action.yml:13]:
```suggestion
  feature_file_path:
```

**Review by Thebarda — APPROVED** [review]

**Review by Thebarda — APPROVED** [review]

**Review by bdauria — APPROVED** [review]

**Review by tuntoja — APPROVED** [review]

**Pull request merged**